src/families/ethereum/nft.integration.test.ts

```diff
@@ -0,0 +1,168 @@
+  id: "js:1:bsc:0xb98d10d9f6d07ba283bfd21b2dfec050f9ae282a:",
+  seedIdentifier: "0xb98d10d9f6d07ba283bfd21b2dfec050f9ae282a",
+  name: "G4sp4rd",
+  derivationMode: "",
+  index: 0,
+  freshAddress: "0xb98d10d9f6d07ba283bfd21b2dfec050f9ae282a",
+  freshAddressPath: "44'/60'/0'/0/0",
+  freshAddresses: [],
+  pendingOperations: [],
+  operations: [],
+  currencyId: "bsc",
+  unitMagnitude: 18,
+  balance: "",
+  blockHeight: 0,
+  lastSyncDate: "",
+  xpub: "",
+};
+
+async function sync(account, withNFT = true) {
+  const bridge = getAccountBridge(account);
+  const blacklistedTokenIds = [];
+  setEnv("NFT", withNFT);
+  const r = await bridge
+    .sync(account, {
+      paginationConfig: {},
+      blacklistedTokenIds,
+    })
+    .pipe(reduce((a, f: (arg0: Account) => Account) => f(a), account))
+    .toPromise();
+  setEnv("NFT", false);
+  return r;
+}
+
+const localCache = {};
+const cache = makeBridgeCacheSystem({
+  saveData(c, d) {
+    localCache[c.id] = d;
+    return Promise.resolve();
+  },
+  getData(c) {
+    return Promise.resolve(localCache[c.id]);
+  },
+});
+
+ethTestingAccounts.forEach((a) => {
+  describe(a.name + " NFT on ethereum", () => {
+    let account = fromAccountRaw(a);
+
+    test("first sync & have nfts", async () => {
+      await cache.prepareCurrency(account.currency);
+      account = await sync(account);
+      expect(account.nfts?.length || 0).not.toBe(0);
+    });
+
+    test("no nft should be on quantity 0 or negative", () => {
+      expect(account.nfts?.find((n) => !n.amount.gt(0))).toEqual(undefined);
+    });
+
+    test("patchAccount restore new NFTs correctly", () => {
+      const copy = {
+        ...account,
+        operations: [],
+        nfts: account.nfts?.slice(Math.ceil((account.nfts?.length || 0) / 2)),
+      };
+      const newAccount = patchAccount(copy, toAccountRaw(account));
+      expect(newAccount.nfts).toEqual(account.nfts);
+    });
+
+    test("remove half NFTs will restore them with half operations", async () => {
+      const halfOps = Math.ceil(account.operations.length / 2);
+      const copy = {
+        ...account,
+        operations: account.operations.slice(halfOps),
+        nfts: account.nfts?.slice(Math.ceil((account.nfts?.length || 0) / 2)),
+      };
+      const resync = await sync(copy);
+      expect(resync.nfts).toEqual(account.nfts);
+    });
+
+    test("start account with .nfts and disable the NFT flag should make it disappear", async () => {
+      expect(account.nfts).not.toBeFalsy();
+      const resync = await sync(account, false);
+      expect(resync.nfts).toBeFalsy();
+    });
+
+    test("start account without .nfts and enable the NFT flag should make it appear", async () => {
+      const first = await sync(account, false);
+      expect(first.nfts).toBeFalsy();
+      const second = await sync(first, true);
+      expect(second.nfts).not.toBeFalsy();
+      expect(second.nfts).toEqual(account.nfts);
+    });
+  });
+});
```

Comment:
we reduced the # of tests to the essential ones to optim. also split into two files.